src/Codeworx.Identity.EntityFrameworkCore.Api/ClientConfigurationController.cs

```diff
@@ -0,0 +1,271 @@
+            }).ToListAsync();
+
+            return configurations;
+        }
+
+        [HttpGet("{id:guid}")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ClientConfigurationInfoData> GetClientConfigurationByIdAsync(Guid id)
+        {
+            var configuration = await _db.Context.Set<ClientConfiguration>().Where(p => p.Id == id).Select(p => new ClientConfigurationInfoData
+            {
+                Id = p.Id,
+                AccessTokenType = p.AccessTokenType,
+                AccessTokenTypeConfiguration = p.AccessTokenTypeConfiguration,
+                AuthenticationMode = p.AuthenticationMode,
+                TokenExpiration = p.TokenExpiration,
+                UserId = p.UserId,
+                ClientType = p.ClientType,
+                Scopes = p.ScopeAssignments.Select(t => t.ScopeId).ToList(),
+                ValidRedirectUrls = p.ValidRedirectUrls.Select(t => t.Url).ToList(),
+            }).FirstOrDefaultAsync();
+
+            if (configuration == null)
+            {
+                // TODO return 404;
+                throw new NotImplementedException();
+            }
+
+            return configuration;
+        }
+
+        [HttpGet("byuser/{userId}")]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IEnumerable<ClientConfigurationInfoData>> GetClientConfigurationByUserIdAsync(Guid userId)
+        {
+            if (!await _db.Context.Set<User>().AnyAsync(p => p.Id == userId))
+            {
+                // TODO return 404;
+                throw new NotImplementedException();
+            }
+
+            var configurations = await _db.Context.Set<ClientConfiguration>().Where(p => p.UserId == userId).Select(t => new ClientConfigurationInfoData
+            {
+                Id = t.Id,
+                AccessTokenType = t.AccessTokenType,
+                AccessTokenTypeConfiguration = t.AccessTokenTypeConfiguration,
+                AuthenticationMode = t.AuthenticationMode,
+                ClientType = t.ClientType,
+                TokenExpiration = t.TokenExpiration,
+                UserId = t.UserId,
+                Scopes = t.ScopeAssignments.Select(t => t.ScopeId).ToList(),
+                ValidRedirectUrls = t.ValidRedirectUrls.Select(t => t.Url).ToList(),
+            }).ToListAsync();
+
+            return configurations;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ClientConfigurationInfoData> InsertClientConfigurationAsync([FromBody] ClientConfigurationInsertData configuration)
+        {
+            var entity = new ClientConfiguration
+            {
+                Id = Guid.NewGuid(),
+                ClientSecretHash = _hashingProvider.Create(configuration.Secret),
```

Comment:
The Client Secret shoud be server generated and returned from the Post call.

Also a HttpPut ResetClientSecret Operation is required.

Only generate clientSecrets for clientTypes  (Web = 0x01,   Backend = 0x08 and  ApiKey = 0x10)

Reply:
done